### ISSUE TITLE
Add `cast_known_wrap` for known bad casts in const contexts ( `cast_possible_wrap`)

### ISSUE THREAD

**Issue by tgross35** [rust-lang/rust-clippy#10299]
### What it does

Casting between signed and unsigned constants is somewhat common, especially when working with C FFI (bindgen typically generates unsigned consts, C functions typically use signed ints).

This lint would check for `as` casts in const contexts and flag any where the RHS does not have the same value as the LHS.

### Lint Name

cast_known_wrap

### Category

pedantic

### Advantage

Catch cases where a C `#define` doesn't fit into the expected value

### Drawbacks

Noise when the cast is intended to just reinterpret the bit value

### Example

```rust
#![warn(clippy::pedantic)]
#![warn(clippy::nursery)]

// OK conversion
const A: u32 = 1;
const B: i32 = A as i32;

// Failed conversion
const X: u32 = u32::MAX;
const Y: i32 = X as i32;
// const Z: i32 = X.try_into().unwrap(); // not possible in const


fn main() {
    println!("{A} {B}");
    println!("{X} {Y}");
}
```

Could be written as:

(no alternative - just a warning)
